### DISCUSSION TITLE
Nuxt 3 route meta fields

### DISCUSSION THREAD

**Discussion by Sin1tar** [nuxt/nuxt#18097] [Questions]
Hello dear developers, please, can you tell me how you set the meta information for the route? I tried to apply the method from the previous version (https://github.com/nuxt/nuxt.js/tree/2.x/examples/routes-meta), however, it is logical that it did not work. Tell me please, how you solve this problem in the third version? Thanks a lot in advance.

Most likely this task at the moment can be solved using this method (https://github.com/nuxt/framework/blob/main/examples/module-extend-pages/modules/pages/index.ts), however, it seems quite large for the current task.

**Comment by viii47** [comment]
I'm assuming you want to set reactive attributes to the body class (as seen in the example), or other classes/attributes for that matter.

From my personal experience with Nuxt3 I'd incorporate `useState()` (see https://v3.nuxtjs.org/docs/usage/state/) to define and set the class/attribute globally and to make it reactive via `useMeta()` in any component/page that you like.

`useMeta` can have static & computed values (see https://github.com/vueuse/head).

**Reply by Sin1tar** [reply]
> I'm assuming you want to set reactive attributes to the body class (as seen in the example), or other classes/attributes for that matter.
> 
> From my personal experience with Nuxt3 I'd incorporate `useState()` (see https://v3.nuxtjs.org/docs/usage/state/) to define and set the class/attribute globally and to make it reactive via `useMeta()` in any component/page that you like.
> 
> `useMeta` can have static & computed values (see https://github.com/vueuse/head).

Thanks a lot for your answer, however I meant route metadata, sorry for the undeveloped content of the my question https://next.router.vuejs.org/guide/advanced/meta.html#route-meta-fields

**Reply by Lord-Y** [reply]
I opened this discussion https://github.com/nuxt/framework/discussions/1936 a while ago but I didn't have any luck so far :snif

**Reply by ceferrari** [reply]
@Lord-Y Did you get an answer for this? I can't open the discussion

**Reply by Lord-Y** [reply]
@ceferrari nope. I can see that the discussion is niw gone 🫤

**Comment by ceferrari** [comment]
More than three years have passed and I am also looking for an answer to this question.

I can pass query:
```
<NuxtLink :to="{ path: 'abc', query: { def: 'def' }}" />
```
I can pass params:
```
<NuxtLink :to="{ path: 'xyz', params: { id: 123 }}" />
```
But I can't pass meta:
```
<NuxtLink :to="{ path: 'my-route', meta: { title: 'My Route' }}" />
```
